recipes/msys2/all/conanfile.py

```diff
@@ -33,12 +30,18 @@ def source(self):
         # source files downloaded will be different based on architecture or OS
         pass
 
-    def build(self):
-        arch = str(self.settings.arch_build)
-        tools.download(**self.conan_data["sources"][self.version][arch])
-        filename = self.conan_data["sources"][self.version][arch]["filename"]
-        sha256 = self.conan_data["checksum"][self.version][arch]["sha256"]
+    def _download(self, url, sha256):
+        from six.moves.urllib.parse import urlparse
+        filename = os.path.basename(urlparse(url).path)
```

Comment:
Can we make this more explicit? the import and getting the name that way seems complex. Is it possible to compose the file name?

Reply:
I am not sure I understand. suggestions are welcomed. here I just extract path component from the URL, e.g. `http://example.com/some/file.tgz?param=value` will result in `some/file.tgz`. then I just get filename itself - `file.tgz`. do you want something different?

Reply:
Nevermind. I thought this was the source method